Better tally + others improvements

KenAlin opened the pull request:
Hi there,
I'm still sorry for my first pull request for TEdit, the one that broke the tally counter (see #720).

I just updated Visual Studio, and I improved the tally at the same time.

Here is how the tally looks like : 
![http://i.imgur.com/La3RJXk.png](http://i.imgur.com/La3RJXk.png)

* Monsters you don't have the banner yet will be printed apart of the
others. (Section "kills", section "less than 50 kills", section "no kill".)
* Reworked algorithm, will explore the monsters list only once now.
* Removed "banner" as suffix.
* For "banner(s) earned" : with an "s" only when >1.

I tested the tally on 5 differents worlds (including 1 with no kill at all). 

## Others improvements
### Additions
* Log now register hour, minutes and seconds (minor).
* Merged some tests when analyzing worlds (minor).
* When the world has finished rendering, the progress bar is now at 100% (minor).
* ~~Optimized some icons.~~

### Bug fixes
* ~~Better *versionning* :~~
  * ~~Fixed v0.0.0.0.~~
  * ~~Update checker works (need your approval, I'm not sure about my edit !!).~~
* Fixed some properties when reading world : 
  * There was a typo : HalloweeenKing -> HalloweenKing
  * MaxRain -> TempMaxRain to match the others properties
* ~~Fixed the tile hovered (in blue) :~~
  * ~~In previous beta versions, the tile above the one hovered was in blue.~~

BinaryConstruct commented:
Area all the icon changes necessary? They don't seem to change much except make the diff log longer and harder to review.

KenAlin commented:
Hi, thanks for review. I reverted somes changes.
As for the assemblyinfo file ... There is some kind of error in it, that's why the title bar says "v0.0.0.0", but I'm not a pro for these things.

(I also updated the changelog above.)

BinaryConstruct commented:
Thank you for the update
